src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -101,32 +117,52 @@ impl<T: 'static> EventLoop<T> {
                 result
             });
 
-        event_loop
-            .handle()
-            .register_dispatcher(wayland_dispatcher.clone())?;
+        map_err!(
+            event_loop
+                .handle()
+                .register_dispatcher(wayland_dispatcher.clone()),
+            WaylandError::Calloop
+        )?;
 
         // Setup the user proxy.
         let pending_user_events = Rc::new(RefCell::new(Vec::new()));
         let pending_user_events_clone = pending_user_events.clone();
         let (user_events_sender, user_events_channel) = calloop::channel::channel();
-        event_loop.handle().insert_source(
-            user_events_channel,
-            move |event, _, winit_state: &mut WinitState| {
-                if let calloop::channel::Event::Msg(msg) = event {
-                    winit_state.dispatched_events = true;
-                    pending_user_events_clone.borrow_mut().push(msg);
-                }
-            },
+        map_err!(
+            event_loop
+                .handle()
+                .insert_source(
+                    user_events_channel,
+                    move |event, _, winit_state: &mut WinitState| {
+                        if let calloop::channel::Event::Msg(msg) = event {
+                            winit_state.dispatched_events = true;
+                            pending_user_events_clone.borrow_mut().push(msg);
+                        }
+                    },
+                )
+                .map_err(|error| error.error),
+            WaylandError::Calloop
```

Comment:
For big expressions like this, I usually prefer to leave it outside of the macro in order to preserver proper formatting. Like so:

```rust
let res = / *... */;
map_err!(res, WaylandError::Calloop)
```